ambry-server/src/integration-test/java/com/github/ambry/server/StatsManagerIntegrationTest.java

```diff
@@ -0,0 +1,167 @@
+    DataNodeId dataNodeId = new MockDataNodeId(Collections.singletonList(new Port(6667, PortType.PLAINTEXT)),
+        Collections.singletonList("/tmp"), "DC1");
+
+    for (int i = 0; i < 6; i++) {
+      String partitionClassName =
+          i % 2 == 0 ? MockClusterMap.DEFAULT_PARTITION_CLASS : MockClusterMap.SPECIAL_PARTITION_CLASS;
+      PartitionId partitionId =
+          new MockPartitionId(i, partitionClassName, Collections.singletonList((MockDataNodeId) dataNodeId), 0);
+      Map<StatsReportType, StatsSnapshot> snapshotsByType = StatsManagerTest.generateRandomSnapshot();
+      StoreStats storeStats = new StatsManagerTest.MockStoreStats(snapshotsByType, false, null);
+      storeMap.put(partitionId, new StatsManagerTest.MockStore(storeStats));
+      replicas.add(partitionId.getReplicaIds().get(0));
+      aggregatedAccountStatsSnapshot.getSubMap()
+          .put(Utils.statsPartitionKey(i), snapshotsByType.get(StatsReportType.ACCOUNT_REPORT));
+      aggregatedPartitionClassStatsSnapshot.getSubMap()
+          .computeIfAbsent(partitionClassName, k -> new StatsSnapshot(0L, new HashMap<>()))
+          .getSubMap()
+          .put(Utils.statsPartitionKey(i), snapshotsByType.get(StatsReportType.PARTITION_CLASS_REPORT));
+    }
+    aggregatedAccountStatsSnapshot.updateValue();
+    aggregatedPartitionClassStatsSnapshot.updateValue();
+    StorageManager storageManager = new MockStorageManager(storeMap, dataNodeId);
+    statsManager = new StatsManager(storageManager, replicas, new MetricRegistry(),
+        new StatsManagerConfig(new VerifiableProperties(properties)), new MockTime(), null, null, null);
+  }
+
+  /**
+   * Clean up the database by removing all the data in all tables.
+   * @throws Exception
+   */
+  @Before
+  public void before() throws Exception {
+    Connection dbConnection = accountStatsMySqlStore.getMySqlDataAccessor().getDatabaseConnection(true);
+    Statement statement = dbConnection.createStatement();
+    for (String table : AccountStatsMySqlStore.TABLES) {
+      statement.executeUpdate("DELETE FROM " + table);
+    }
+  }
+
+  /**
+   * Test account stats publisher
+   * @throws Exception
+   */
+  @Test
+  public void testAccountStatsPublisher() throws Exception {
+    StatsManager.AccountStatsPublisher publisher = statsManager.new AccountStatsPublisher(accountStatsMySqlStore);
+    publisher.run();
+
+    StatsWrapper statsWrapper = accountStatsMySqlStore.queryAccountStatsByHost(HOSTNAME);
+    assertEquals(aggregatedAccountStatsSnapshot, statsWrapper.getSnapshot());
+  }
+
+  /**
+   * Test partition class stats publisher
+   * @throws Exception
+   */
+  @Test
+  public void testPartitionClassStatsPublisher() throws Exception {
+    // Before publishing partition class stats, we have to publish account stats since they use the same data
+    StatsManager.AccountStatsPublisher accountStatsPublisher =
+        statsManager.new AccountStatsPublisher(accountStatsMySqlStore);
```

Comment:
Should also test this will null store since that is an option.

Reply:
This integration test is mainly for testing mysql database. In unit test, we consider the mysql database doesn't exist by default. So we have to do the test for mysql database here. The case when accountStatsMysqlStore is null, has already been tested in unit tests.